src/app/find/errorReports/ErrorReportContainer.jsx

```diff
@@ -0,0 +1,187 @@
+    super(props);
+    this.state = {
+      errorReportInformation: {
+        errorReportGeneralLocationError: false,
+        errorReportServices: [],
+      },
+      errorReportText: '',
+    };
+
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation();
+    }
+  }
+
+  onErrorReportTextChanged(event) {
+    this.setState({ errorReportText: event.target.value });
+  }
+
+  onErrorReportServicesChanged(value) {
+    const newErrorReportInformation = { ...this.state.errorReportInformation };
+
+    const newErrorReportServices = newErrorReportInformation.errorReportServices.includes(value)
+      ? newErrorReportInformation.errorReportServices.filter(service => service !== value)
+      : [...newErrorReportInformation.errorReportServices, value];
+
+    newErrorReportInformation.errorReportServices = newErrorReportServices;
+
+    this.setState({
+      errorReportInformation: newErrorReportInformation,
+    });
+  }
+
+  onErrorReportGeneralLocationErrorChanged(value) {
+    const newErrorReportInformation = { ...this.state.errorReportInformation };
+
+    newErrorReportInformation.errorReportGeneralLocationError = value;
+
+    this.setState({
+      errorReportInformation: newErrorReportInformation,
+    });
+  }
+
+  onErrorReportSubmitted() {
+    const data = {
+      generalLocationError: this.state.errorReportInformation.errorReportGeneralLocationError,
+      services: this.state.errorReportInformation.errorReportServices,
+    };
+
+    if (this.state.errorReportText !== '') {
+      data.content = this.state.errorReportText;
+    }
+
+    this.props.postErrorReport(data);
+  }
+
+  render() {
+    const {
+      locationData,
+      locationError,
+      goToViewLocation,
+    } = this.props;
+
+    const {
+      errorReportGeneralLocationError,
+      errorReportServices,
+    } = this.state.errorReportInformation;
+
+    if (locationError) {
+      return <ErrorLabel errorMessage={locationError} />;
+    }
+
+    if (!locationData) {
+      return <LoadingView message="Loading location..." />;
+    }
+
+    return (
+      <Router>
+        <div>
+          <Modal>
+            <div className="mx-3 mt-4 position-relative">
+              <Icon
+                name="times"
+                onClick={goToViewLocation}
+                style={{
+                  position: 'absolute',
+                  right: 0,
+                  top: '0.2em',
+                }}
+              />
+            </div>
+            <Switch>
+              <Redirect exact from={`${this.props.match.path}/`} to={`${this.props.match.path}/services`} />
```

Comment:
Is the redirection actually needed? Why not just make services be at `/` in the first place?
(But this isn't really an issue, unless it affects back navigation in some weird way, which it probably doesn't.)

Reply:
I did consider services just being at `/`, but thought I'd better give it it's own named endpoint.
Can change if you'd like, or leave for a refactor later.